tests/install_test.go

```diff
@@ -0,0 +1,91 @@
+package tests
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"os/user"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"code.gitea.io/gitea/tests/internal/utils"
+)
+
+// The HTTP port listened by the Gitea server.
+const ServerHTTPPort = "3001"
+
+const _RetryLimit = 10
+
+func makeSimpleSettings(user, workdir, port string) map[string][]string {
+	return map[string][]string{
+		"db_type":        {"SQLite3"},
+		"db_host":        {"localhost"},
+		"db_path":        {workdir + "data/gitea.db"},
+		"app_name":       {"Gitea: Git with a cup of tea"},
+		"repo_root_path": {workdir + "repositories"},
+		"run_user":       {user},
+		"domain":         {"localhost"},
+		"ssh_port":       {"22"},
+		"http_port":      {port},
+		"app_url":        {"http://localhost:" + port},
+		"log_root_path":  {workdir + "log"},
+	}
+}
+
+func install(t *utils.T) error {
+	var r *http.Response
+	var err error
+
+	for i := 1; i <= _RetryLimit; i++ {
+
+		r, err = http.Get("http://:" + ServerHTTPPort + "/")
+		if err == nil {
+			fmt.Fprintln(os.Stderr)
+			break
+		}
+
+		// Give the server some amount of time to warm up.
+		time.Sleep(100 * time.Millisecond)
+		fmt.Fprintf(os.Stderr, ".")
```

Comment:
No, I meant keep the sleep, but replace the call to `fmt.Fprintf(..)` (line 50) with `fmt.Fprint(..)`. No reason to use `fmt.Fprintf(..)` if you're not using any formatting. Sorry for any confusion.